### DISCUSSION TITLE
Apply transforms twice?

### DISCUSSION THREAD

**Discussion by luxunxiansheng** [Project-MONAI/MONAI#5769] [Q&A]
Hi, there,

Suppose I have 100 image files and use the CacheDataset with a cache rate 0.5  and set it to the train laoder 
When training , to get batchs like this:
      for batch_data in tranin_Loader:
        ....

I notice that the line above is very slow. It seems to apply_transforms again besides constructing the cachedataset. 

I missed something?

Thanks in advacne
Bin

**Comment by KumoLiu** [comment]
Hi @luxunxiansheng,

Not sure what you mean, you can use `CacheDataset` like this:
```
train_ds = CacheDataset(
    data=train_files, transform=train_transforms,
    cache_rate=0.5, num_workers=4)
train_loader = DataLoader(train_ds, batch_size=2, shuffle=True, num_workers=4)

for batch_data in train_loader:
.....
```
You can also refer to this [tutorial](https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/spleen_segmentation_3d.ipynb)

Hope it can help you, thanks!
